src/Microsoft.Health.Dicom.Core/Features/Common/DicomTagParser.cs

```diff
@@ -27,24 +28,79 @@ public bool TryParse(string dicomTagPath, out DicomTag[] dicomTags, bool support
                 return false;
             }
 
+            DicomTag dicomTag = GetTagFromKeywordOrNumber(dicomTagPath);
 
+            dicomTags = new DicomTag[] { dicomTag };
+            return dicomTag != null;
+        }
+
+        public bool TryParseToDicomItem(string dicomTagPath, out DicomItem dicomItem)
+        {
+            dicomItem = null;
+
+            if (string.IsNullOrWhiteSpace(dicomTagPath))
+            {
+                return false;
+            }
+
+            DicomTag[] dicomTags = ParseMultipleStandardTag(dicomTagPath);
+
+            if (dicomTags.Length == 1)
+            {
+                var tag = GetTagFromKeywordOrNumber(dicomTagPath);
+                if (tag != null) dicomItem = new DicomValuelessItem(tag);
+                return tag != null;
+            }
+
+            if (dicomTags.Length > 2)
+            {
+                throw new NotImplementedException(DicomCoreResource.NestedSequencesNotSupported);
+            }
+
+            if (dicomTags[0] != null && dicomTags[1] != null)
+            {
+                var sequenceItem = new DicomSequence(dicomTags[0]);
+                sequenceItem.Items.Add(new DicomDataset(new DicomValuelessItem(dicomTags[1])));
+
+                dicomItem = sequenceItem;
+            }
+
+
+            return dicomTags[0] != null && dicomTags[1] != null;
+        }
+
+
+        private static DicomTag[] ParseMultipleStandardTag(string dicomTagPath)
+        {
+            var paths = dicomTagPath.Split(".");
+            var dicomTags = new List<DicomTag>();
+
+            foreach (var path in paths)
+            {
+                dicomTags.Add(GetTagFromKeywordOrNumber(path));
```

Comment:
Do we want to handle case where `GetTagFromKeywordOrNumber` returns `null`? which indicates the input is invalid.

Reply:
It is handled in the parent method, where we check for if it is not null. If any one of them is null we return false for parsing the elements. A unit test also covers that scenario